File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/typography/TypographyRoles.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.typography

import com.perrystreet.woof.designsystem.atomic._primitives.typography.TypographyPrimitives

object TypographyRoles {
    val Default = Typography(
        display = Typography.Display(
            h1 = TypographyPrimitives.Bold.Size32,
            h2 = TypographyPrimitives.Bold.Size24,
            h3 = TypographyPrimitives.Bold.Size20,
            h4 = TypographyPrimitives.Bold.Size16,
            h5 = TypographyPrimitives.Bold.Size14,
            h6 = TypographyPrimitives.Bold.Size12,
        ),
        subhead = Typography.Subhead(
            p1 = TypographyPrimitives.SemiBold.Size16,
            p2 = TypographyPrimitives.SemiBold.Size14,
            p3 = TypographyPrimitives.SemiBold.Size12,
        ),
        body = Typography.Body(
            p1 = TypographyPrimitives.Regular.Size16,
            p2 = TypographyPrimitives.Regular.Size14,
            p3 = TypographyPrimitives.Regular.Size12,
            p4 = TypographyPrimitives.Regular.Size10,
        ),
    )
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Typography/TypographyRoles.swift
import SwiftUI

public enum TypographyRoles {
    public static let `default` = Typography(
        display: Typography.Display(
            h1: TypographyPrimitives.Bold.size32,
            h2: TypographyPrimitives.Bold.size24,
            h3: TypographyPrimitives.Bold.size20,
            h4: TypographyPrimitives.Bold.size16,
            h5: TypographyPrimitives.Bold.size14,
            h6: TypographyPrimitives.Bold.size12
        ),
        subhead: Typography.Subhead(
            p1: TypographyPrimitives.SemiBold.size16,
            p2: TypographyPrimitives.SemiBold.size14,
            p3: TypographyPrimitives.SemiBold.size12
        ),
        body: Typography.Body(
            p1: TypographyPrimitives.Regular.size16,
            p2: TypographyPrimitives.Regular.size14,
            p3: TypographyPrimitives.Regular.size12,
            p4: TypographyPrimitives.Regular.size10
        )
    )
}
